tests/foreman/cli/test_rhcloud_inventory.py

```diff
@@ -107,10 +112,26 @@ def test_positive_inventory_recommendation_sync():
 
     :BZ: 1957186
 
-    :CaseAutomation: NotAutomated
+    :CaseAutomation: Automated
 
     :CaseLevel: System
     """
+    org, ak = organization_ak_setup
+    cmd = f'organization_id={org.id} foreman-rake rh_cloud_insights:sync'
+    timestamp = datetime.utcnow().strftime('%Y-%m-%d %H:%M')
+    result = rhcloud_sat_host.execute(cmd)
+    wait_for(
+        lambda: rhcloud_sat_host.api.ForemanTask()
+        .search(query={'search': f'Insights full sync and started_at >= "{timestamp}"'})[0]
+        .result
+        == 'success',
+        timeout=400,
+        delay=15,
+        silent_failure=True,
+        handle_exception=True,
+    )
+    assert result.status == 0
+    assert result.stdout == 'Synchronized Insights hosts hits data\n'
```

Comment:
You are referring to the result before inventory sync is completed or just started!  You might have status code `0` but it would be for start syncing but not for completion. I recommend you to get this status for sync completion else you will have false pass results.

Also, stdout is correct here ? and is it for start and not for complete? if not for complete, check it for complete to assert the sync works !

Reply:
In the `wait_for` function there is a small assertion kind of comparison 'result=='sucess'' that takes care of whether the task resulted in success as the cmd got executed previously, it's just checking the status. About the stdout you are right it gets reflected after the completion of the task when the hosts hit insights data.